Sources/JavaScriptKit/BasicObjects/JSPromise.swift

```diff
@@ -52,7 +52,10 @@ public final class JSPromise<Success, Failure>: ConvertibleToJSValue, Constructi
     /** Schedules the `success` closure to be invoked on sucessful completion of `self`.
     */
     public func then(success: @escaping () -> ()) {
-        let closure = JSClosure { _ in success() }
+        let closure = JSClosure { _ in
```

Comment:
All promise closures can be one-shot, so I’m pretty sure we don’t need to hold on to them manually on the Swift side.